[Jared Lim] iP

jaredlhf opened the pull request:
# JaredDeluxe
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

JaredDeluxe frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/jaredlhf/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jaredlhf committed:
Added Greet, Echo, Exit

jaredlhf committed:
implemented Add functionality

jaredlhf committed:
implemented List functionality

jaredlhf committed:
added Task class

jaredlhf committed:
implemented done functionality

jaredlhf committed:
implemented Todo class

jaredlhf committed:
implemented Deadline class

jaredlhf committed:
implemented Event class

jaredlhf committed:
added input file for automated UI testing

jaredlhf committed:
added Expected file for automated UI testing

jaredlhf committed:
completed automated text UI testing

jaredlhf committed:
created DukeExceptions and Handle errors

jaredlhf committed:
added DukeExceptions file

jaredlhf committed:
added Error handling

jaredlhf committed:
refactor Error handling

jaredlhf committed:
implemented delete function

jaredlhf committed:
add JavaDoc

jaredlhf committed:
add Save

jaredlhf committed:
add Dates and Times

jaredlhf committed:
Merge branch 'branch-Level-7'
merge level 7 to master

jaredlhf committed:
Merge 'branch-Level-8'

jaredlhf committed:
Refactor code for more OOP

jaredlhf committed:
Divide classes into packages

jaredlhf committed:
Update gitignore and Fix bug

jaredlhf committed:
Add JUnit tests

jaredlhf committed:
Complete A-Jar

jaredlhf committed:
Add JavaDoc

jaredlhf committed:
Fix parsing bug

jaredlhf committed:
Fix more bugs

jaredlhf committed:
Cleanup code to follow coding standard

jaredlhf committed:
merge master to branch-A-CodingStandard

jaredlhf committed:
Merge branch 'master' into branch-A-JavaDoc

jaredlhf committed:
add Find

jaredlhf committed:
Merge branch 'branch-Level-9'

jaredlhf committed:
Merge branch 'branch-A-JavaDoc'

jaredlhf commented:
# JaredDeluxe
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

JaredDeluxe frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/jaredlhf/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

jaredlhf committed:
Merge branch 'add-gradle-support'

nwhloo99 commented on src/main/java/Jared.java:
```diff
@@ -0,0 +1,49 @@
+     * Constructor for the Jared Class.
+     * @param filePath file path where the data of the task list resides.
+     */
+    public Jared(String filePath) {
```
Any reason why you used filePath here instead of fileName?

nwhloo99 commented on src/main/java/jared/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+ * Deals with data from the file.
+ */
+public class Storage {
+    private final String fileName;
```
Should this be FILE_NAME since its declared final?

nwhloo99 commented on src/main/java/jared/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+            Scanner reader = new Scanner(f);
+            while (reader.hasNext()) {
+                String next = reader.nextLine();
+                String type = Parser.parseData(next, "type");
```
I like the naming here, it is very intuitive. 👍

nwhloo99 commented on src/main/java/jared/task/TaskList.java:
```diff
@@ -0,0 +1,183 @@
+     * @param next Full user input string.
+     * @throws DukeException If user input command is invalid.
+     */
+    public void add(String command, String next) throws DukeException {
```
Any reason why you use next instead of something like fullCommand?

nwhloo99 commented on src/main/java/jared/ui/Ui.java:
```diff
@@ -0,0 +1,64 @@
+ * Deals with user interactions.
+ */
+public class Ui {
+    private final Scanner scan;
```
Should this be named scanner instead?

nwhloo99 reviewed (commented):
I like your code as it is quite intuitive and easy to read! Just pointed out some minor parts where I am not as clear about the naming. Otherwise, I think it is a pretty good job! 👍

jaredlhf committed:
add checkstyle

jaredlhf committed:
Complete A-Gradle

jaredlhf committed:
Merge branch 'branch-A-CheckStyle'

jaredlhf committed:
Add checkstyle suppressions file

jaredlhf committed:
fix gradle build bug

jaredlhf committed:
Setup Javafx and Launcher

jaredlhf committed:
Add basic GUI

jaredlhf committed:
Add Dialog Box

jaredlhf committed:
Add logic to GUI

jaredlhf committed:
Add FXML

Add FXML

ruoyann commented on src/main/java/jared/task/Deadline.java:
```diff
@@ -0,0 +1,62 @@
+    @Override
+    public String toString() {
+        String res = "[D]" + super.toString() + " (by: "
+                + date.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
```
the format of "d MMM" doesn't seem very intuitive to me, can consider changing to "dd MM"

ruoyann commented on src/main/java/jared/task/Deadline.java:
```diff
@@ -0,0 +1,62 @@
+        String res = "[D]" + super.toString() + " (by: "
+                + date.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+        if (time != null) {
+            res += " " + time.format(DateTimeFormatter.ofPattern("H:mm"));
```
the format of "H:mm" doesn't seem intuitive either, can consider "HH:mm"

jaredlhf committed:
Rename shadowJar file naming and update gitignore

jaredlhf committed:
Add assertions to document important assumptions

Assertions are important to verify assumptions about the code.
Add assertions in toString() and saveFormat() functions to ensure that
the string returned is not null to prevent saving of empty strings in
the storage.

jaredlhf committed:
Merge pull request #1 from jaredlhf/branch-A-Assertions

A-Assertions

jaredlhf committed:
Refactor code and improve code quality

Some methods were long and complicated. It was hard to read
due to deep nesting.

Refactor by extracting methods into smaller more concise methods
to improve readability. Rename variable names and method parameters
to more intuitive names.

jaredlhf committed:
Merge branch 'master' into branch-A-CodeQuality

jaredlhf committed:
Merge pull request #2 from jaredlhf/branch-A-CodeQuality

A-CodeQuality

jaredlhf committed:
Add sort function

jaredlhf committed:
add help command

jaredlhf committed:
fix Welcome message

jaredlhf committed:
Refactor code for add command

jaredlhf committed:
Returns error encountered as output

jaredlhf committed:
Improve GUI

jaredlhf committed:
Add product screenshot

jaredlhf committed:
Update README.md for user guide

jaredlhf committed:
Fix link to headers in README.md

jaredlhf committed:
Fix link for exit command

jaredlhf committed:
Update User Guide

jaredlhf committed:
Resolve issue where text box truncates

jaredlhf committed:
Update User Guide

jaredlhf committed:
Merge branch 'master' of https://github.com/jaredlhf/ip

jaredlhf committed:
Refactor code
